exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Table.kt

```diff
@@ -1651,14 +1653,19 @@ open class Table(name: String = "") : ColumnSet(), DdlAware {
                         }
                         is SQLServerDialect -> checkConstraints.filterNot { (name, _) ->
                             name.startsWith("${generatedUnsignedCheckPrefix}byte_") ||
-                                name.startsWith(generatedSignedCheckPrefix)
+                                name.startsWith("${generatedSignedCheckPrefix}short")
                         }
                         is PostgreSQLDialect -> checkConstraints.filterNot { (name, _) ->
-                            name.startsWith(generatedSignedCheckPrefix)
+                            name.startsWith("${generatedSignedCheckPrefix}short")
                         }
                         is H2Dialect -> {
                             when (dialect.h2Mode) {
-                                H2Dialect.H2CompatibilityMode.Oracle -> checkConstraints
+                                H2Dialect.H2CompatibilityMode.Oracle -> checkConstraints.filterNot { (name, _) ->
+                                    name.startsWith("${generatedSignedCheckPrefix}byte")
```

Comment:
Several lines before the signed check for byte was with underscore in the end (`"${generatedUnsignedCheckPrefix}byte_"`), if it's intended it's fine. 

But probably it would be also better to move this strings to constants like `generatedUnsignedCheckPrefix` and `generatedSignedCheckPrefix`

Reply:
This has been simplified in the second commit.